crates/ruff_formatter/src/printer/mod.rs

```diff
@@ -47,20 +47,40 @@ impl<'a> Printer<'a> {
         }
     }
 
-    /// Prints the passed in element as well as all its content
-    pub fn print(self, document: &'a Document) -> PrintResult<Printed> {
-        self.print_with_indent(document, 0)
+    #[must_use]
+    pub fn with_indentation(self, indentation: &str) -> Self {
```

Comment:
I had to change the printer to accept an arbitrary indentation to support `<space><tab><space>` indentation (and not just `<indent><space>`